### PR TITLE
NativeDisplay: Add GenerateLayerHash method

### PR DESCRIPTION
**Pull request by harishkrupo**
The generate layer hash can be used to generate unique hash values
for the layer id.

Jira: None.
Test:

Signed-off-by: Harish Krupo <harish.krupo.kps@intel.com>
Signed-off-by: Kalyan Kondapally <kalyan.kondapally@intel.com>

### CODE REVIEW

**Comment by harishkrupo** [comment]
@kalyankondapally I wasn't fully convinced that only memory addresses (and the ints) we would ensure unique values. So also taking into account the previous VBlank timestamp, whenever available.

File: os/android/iahwc2.cpp [hunk 1]
```diff
@@ -395,10 +401,12 @@ HWC2::Error IAHWC2::HwcDisplay::AcceptDisplayChanges() {
 
 HWC2::Error IAHWC2::HwcDisplay::CreateLayer(hwc2_layer_t *layer) {
   supported(__func__);
+  layer_idx_ = display_.GenerateLayerHash(layer_idx_);
+  layer_idx_ += timestamp_;
   layers_.emplace(static_cast<hwc2_layer_t>(layer_idx_), IAHWC2::Hwc2Layer());
   layers_.at(layer_idx_).XTranslateCoordinates(display_->GetXTranslation());
   *layer = static_cast<hwc2_layer_t>(layer_idx_);
-  ++layer_idx_;
+  layer_idx_ = (uint64_t)&layers_.at(layer_idx_);
```
Comment (kalyankondapally) [inline os/android/iahwc2.cpp]: just for clarity can u change this to (uint64_t)&(layers_.back().second) ?

Comment (harishkrupo) [reply os/android/iahwc2.cpp]: But maps don't seem to have a back() method? [API Doc](http://www.cplusplus.com/reference/map/map/)

File: os/android/iahwc2.cpp [hunk 2]
```diff
@@ -395,10 +401,12 @@ HWC2::Error IAHWC2::HwcDisplay::AcceptDisplayChanges() {
 
 HWC2::Error IAHWC2::HwcDisplay::CreateLayer(hwc2_layer_t *layer) {
   supported(__func__);
+  layer_idx_ = display_.GenerateLayerHash(layer_idx_);
```
Comment (Kishore409) [inline os/android/iahwc2.cpp]: should it be display_->GenerateLayerHash

Comment (harishkrupo) [reply os/android/iahwc2.cpp]: Yeah, thanks. 👍

File: os/android/iahwc2.cpp (hunk 1 above)
Comment (kalyankondapally) [reply os/android/iahwc2.cpp]: Sorry missed that.

**Comment by harishkrupo** [comment]
Closing this in favor of #224

**Pull request closed without merging**